okio-zipfilesystem/src/jvmTest/kotlin/okio/zipfilesystem/FixedLengthSourceTest.kt

```diff
@@ -0,0 +1,132 @@
+/*
+ * Copyright (C) 2021 Square, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package okio.zipfilesystem
+
+import okio.Buffer
+import okio.IOException
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.Test
+import kotlin.test.fail
+
+internal class FixedLengthSourceTest {
```

Comment:
Thinking about a test case along the lines of:
1. Source of length 16, wrap with non-truncated fixed size of 10
2. Read 10 (`bytesReceived` now equal to fixed size)
3. Read 10 and expect error and no data

All the existing tests look to put `bytesReceived` over the size and never equal. I don't think there is a bug or anything, but it might be nice to have a test for this boundary condition.

Reply:
Good call! Added this fencepost test.